examples/ola-rdm.cpp

```diff
@@ -269,44 +282,59 @@ bool RDMController::Setup() {
 /**
  * Handle the RDM response
  */
-void RDMController::HandleResponse(
-    const ola::rdm::ResponseStatus &response_status,
-    const string &rdm_data) {
-  if (!response_status.error.empty()) {
-    cerr << "Error: " << response_status.error << endl;
+void RDMController::HandleResponse(const ola::client::Result &result,
+                                   const ola::client::RDMMetadata &metadata,
+                                   const ola::rdm::RDMResponse *response) {
+  if (!result.Success()) {
+    cerr << "Error: " << result.Error() << endl;
     m_ola_client.GetSelectServer()->Terminate();
     return;
   }
 
-  if (response_status.response_code == ola::rdm::RDM_WAS_BROADCAST) {
+  if (metadata.response_code == ola::rdm::RDM_WAS_BROADCAST) {
     m_ola_client.GetSelectServer()->Terminate();
     return;
-  } else if (response_status.response_code != ola::rdm::RDM_COMPLETED_OK) {
+  } else if (metadata.response_code != ola::rdm::RDM_COMPLETED_OK) {
     cerr << "Error: "
-         << ola::rdm::StatusCodeToString(response_status.response_code) << endl;
+         << ola::rdm::StatusCodeToString(metadata.response_code) << endl;
     m_ola_client.GetSelectServer()->Terminate();
     return;
   }
 
-  if (response_status.response_type == ola::rdm::RDM_ACK_TIMER) {
-    m_ola_client.GetSelectServer()->RegisterSingleTimeout(
-      response_status.AckTimer(),
-      ola::NewSingleCallback(this, &RDMController::FetchQueuedMessage));
+  if (!response) {
+    cerr << "Error: Missing RDM Response but response_code was "
+            "RDM_COMPLETED_OK, this is a bug, please report it!"
+         << endl;
     return;
   }
 
-  if (response_status.response_type == ola::rdm::RDM_ACK) {
-    if (response_status.pid_value == m_pending_request.pid_value ||
+  if (response->ResponseType() == ola::rdm::RDM_ACK_TIMER) {
```

Comment:
But we could add a method to fetch the backoff time in ms directly, rather than each client calculating it (and avoid having the data size validation locally in each client).

I guess the other bits vary too much between clients.